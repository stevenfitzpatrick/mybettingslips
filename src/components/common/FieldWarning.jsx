import PropTypes from 'prop-types';
import React from 'react';

const propTypes = {
  field: PropTypes.string.isRequired,
  touched: PropTypes.object,
  errors: PropTypes.object
};

const defaultProps = {
  touched: {},
  errors: {}
};

const FieldWarning = ({ touched, errors, field }) => {
  return touched[field] && errors[field] ? (
    <div className="form-error">{errors[field]}</div>
  ) : null;
};

FieldWarning.propTypes = propTypes;
FieldWarning.defaultProps = defaultProps;

export default FieldWarning;
