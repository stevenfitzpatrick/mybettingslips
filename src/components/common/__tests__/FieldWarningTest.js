import React from 'react';
import renderer from 'react-test-renderer';

import FieldWarning from '../FieldWarning';

describe('Field Warning', () => {
  const defaultProps = {
    field: 'name',
    errors: {},
    touched: {}
  };

  const defaultPropsWithError = {
    ...defaultProps,
    errors: { name: 'Name is Required' },
    touched: { name: true }
  };

  test('should not render without error and touched', () => {
    const tree = renderer.create(<FieldWarning {...defaultProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('should not render with error and touched', () => {
    const tree = renderer
      .create(<FieldWarning {...defaultPropsWithError} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
