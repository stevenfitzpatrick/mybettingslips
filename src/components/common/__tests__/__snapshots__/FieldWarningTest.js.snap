// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Field Warning should not render with error and touched 1`] = `
<div
  className="form-error"
>
  Name is Required
</div>
`;

exports[`Field Warning should not render without error and touched 1`] = `null`;
